packages/plugins/plugin-markdown/src/capabilities/intent-resolver.ts

```diff
@@ -48,44 +47,23 @@ export default (context: PluginsContext) =>
     // TODO(burdon): What is the error boundary for intents? Are errors reported back to caller?
     createResolver({
       intent: CollaborationActions.InsertContent,
-      resolve: async ({ label, queueId, messageId, associatedArtifact }) => {
+      resolve: async ({ spaceId, target: targetRef, object: objectRef, label }) => {
         const client = context.requestCapability(ClientCapabilities.Client);
-        const layout = context.requestCapability(Capabilities.Layout);
-
-        // Only handle markdown documents.
-        const { id, typename } = associatedArtifact;
-        log('processing proposal', { queueId, messageId, associatedArtifact });
-        if (typename !== DocumentType.typename) {
-          log.warn('invalid object type', { associatedArtifact });
-          return;
-        }
-
-        // Get the document from the associatedArtifact.
-        let document;
-        const { spaceId } = parseId(layout.workspace);
-        const space = spaceId ? client.spaces.get(spaceId) : null;
-        if (space) {
-          const [objectSpaceId, objectId] = parseFullyQualifiedId(id);
-          if (objectSpaceId !== spaceId) {
-            log.warn('invalid space', { spaceId, associatedArtifact });
-            return;
-          }
-
-          document = await space.db.query({ id: objectId }).first();
-        }
-        if (!document) {
-          log.warn('document not found', { associatedArtifact });
-          return;
+        const space = client.spaces.get(spaceId);
+        const target = await resolveRef(client, targetRef.dxn, space);
+        if (target && isInstanceOf(DocumentType, target)) {
+          const accessor = createDocAccessor(target, ['content']);
+          // TODO(burdon): Should be a cursor that references a selected position.
+          const index = 0;
+          accessor.handle.change((doc) => {
+            // TODO(burdon): Throws error:
+            // intent-dispatcher.ts:270 Cannot read properties of undefined (reading 'annotations') (FiberFailure) TypeError: Cannot read properties of undefined (reading 'annotations')
+            const ref = `[${label ?? 'Generated content'}]](${objectRef.dxn.toString()})\n`;
+            A.splice(doc, accessor.path.slice(), index, 0, ref);
```

Comment:
@dmaretskyi this is currently throwing

Reply:
mutating the am doc needs to be within a `change` doesn't it?